usr/src/lib/brand/bhyve/zone/boot.c

```diff
@@ -450,14 +454,27 @@ add_nets(int *argc, char **argv)
 			}
 			primary = net;
 			pcifn = 0;
+		} else if ((slotstr = get_zcfg_var("net", net,
+		    "pci_slot")) != NULL) {
+			if (parse_pcislot(slotstr, &pcibus, &pcislot,
+			    &pcifn) != 0) {
+				return (-1);
+			}
 		} else {
 			pcifn = nextpcifn;
 			nextpcifn++;
 		}
 
-		if (snprintf(slotconf, sizeof (slotconf),
-		    "%d:%d,virtio-net-viona,%s", PCI_SLOT_NICS, pcifn, net) >=
-		    sizeof (slotconf)) {
+		if (pcibus > 0) {
```

Comment:
I see no way to avoid collisions here?

I can add a nic without pci_slot set and mark it primary and then add a 2nd one set to 0:6:0 an they will end up in the same slot. (Or am I missing a check for that somewhere?